File: leetcode/day5/cpp/main.cpp

#include "so.h"
#include <iostream>
#include <string>
#include <vector>
#include <stdio.h>

int main(int argc, char **argv)
{
	Solution s{};

	ListNode *l = s.addTwoNumbers(new ListNode(2, new ListNode(4, new ListNode(3))),
				      new ListNode(5, new ListNode(6, new ListNode(4))));

	ListNode *_l = l;
	printf("ssss\n");
	while (_l != nullptr)
	{
		printf("the value is %d \n", _l->val);
		// std::cout << _l->val << std::endl;
		_l = _l->next;
	}

	return 0;
}

File: leetcode/day5/python/main.py

import so
from so import ListNode
from typing import Optional


def main():

    s = so.Solution()

    l1 = ListNode(2, ListNode(4, ListNode(3)))

    l2 = ListNode(5, ListNode(6, ListNode(4)))

    l3 = s.addTwoNumbers(l1, l2)

    _l = l3

    while (_l != None):
        print(_l.val)
        _l = _l.next


if __name__ == "__main__":
    main()
